tests/unit/s2n_fork_generation_number_test.c

```diff
@@ -330,6 +330,11 @@ int main(int argc, char **argv)
 
     EXPECT_TRUE(s2n_array_len(fgn_test_cases) == NUMBER_OF_FGN_TEST_CASES);
 
+    /* Sanity check that MAP_INHERIT_ZERO is supported where we know it should be supported */
+#if defined(__FreeBSD__) || defined(__OpenBSD__)
```

Comment:
I was trying to avoid making assumptions on only the OS. Hence, feature probing for existence of the function `minherit()` and then checking (via the macro-is-defined conditional) that the functions supports the feature.

Reply:
My goal wasn't to make a complete check here (like, we don't need to call out MacOS and all other OSes here), but to just check two cases where we know the answer. This seems important, since this test would have failed with the old code.